src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -188,6 +188,11 @@ internal JsonTypeInfo(Type type, JsonConverter converter, JsonSerializerOptions
                     ThrowHelper.ThrowInvalidOperationException_JsonTypeInfoOperationNotPossibleForKind(Kind);
                 }
 
+                if (Type.IsImmutableEnumerableType() || Type.IsImmutableDictionaryType())
```

Comment:
These helpers use reflection, which is not ideal when those setters are being invoked by the source generator. This should be specified by the converter instead.

Reply:
Done.